junit-plugin/src/main/scala/org/scalajs/junit/plugin/ScalaJSJUnitPlugin.scala

```diff
@@ -0,0 +1,447 @@
+          mkInvokeJUnitMethodDef(annotatedMethods, localMethodSymbol)
+        }
+
+        val newBody = getJUnitMetadataDef :: newInstanceDef :: invokeJUnitMethodDef :: Nil
+
+        val newParents = TypeTree(typeOf[java.lang.Object]) ::
+          TypeTree(typeOf[JUnitTestMetadata]) :: Nil
+
+        val newImpl = clazz.impl.copy(parents = newParents, body = newBody)
+
+        val newClazz = {
+          val newClassName = newTypeName(clazz.name.toString + "$scalajs$junit$hook")
+          val newClazz = gen_mkClassDef_compat(Modifiers(Flags.MODULE),
+            newClassName, Nil, newImpl)
+          clazzSym.flags += Flags.MODULE
+          clazzSym.withoutAnnotations
+          clazzSym.setName(newClassName)
+          val newClazzInfo = {
+            val newParentsInfo = typeOf[java.lang.Object] ::
+              typeOf[JUnitTestMetadata] :: Nil
+            val decls = clazzSym.info.decls
+            decls.enter(getJUnitMetadataDef.symbol)
+            decls.enter(newInstanceDef.symbol)
+            decls.enter(invokeJUnitMethodDef.symbol)
+            ClassInfoType(newParentsInfo, decls, clazzSym.info.typeSymbol)
+          }
+          clazzSym.setInfo(newClazzInfo)
+          scalaJSPlugin.PrepInteropComponent.registerModuleExports(clazzSym)
+
+          newClazz.setSymbol(clazzSym)
+          newClazz
+        }
+        typer.typedClassDef(newClazz)
+      }
+
+      def jUnitAnnotatedMethods(clDef: ClassDef): List[DefDef] = {
+        clDef.impl.body.collect {
+          case ddef: DefDef if hasJUnitMethodAnnotation(ddef) => ddef
+        }
+      }
+
+      def getMethods(clazzDef: ClassDef): List[DefDef] =
+        clazzDef.impl.body.collect { case dDef: DefDef => dDef }
+
+      def mkInvokeJUnitMethodDef(methods: List[DefDef],
+          localMethodSymbol: Symbol): DefDef = {
+        val invokeJUnitMethodSym = localMethodSymbol.cloneSymbol
+        invokeJUnitMethodSym.withoutAnnotations
+        invokeJUnitMethodSym.setName(newTermName("scalajs$junit$invoke"))
+
+        val paramSym =
+          invokeJUnitMethodSym.newValueParameter(newTermName("methodId"))
+        paramSym.setInfo(typeOf[java.lang.String])
+
+        invokeJUnitMethodSym.setInfo(MethodType(List(paramSym), typeOf[Unit]))
+
+        val methodNotFound = {
+          val msg = gen.mkMethodCall(paramSym, newTermName("$plus"), Nil,
+            List(Literal(Constant(" not found"))))
+          val exceptionConstructor = Select(
+            New(TypeTree(typeOf[NoSuchMethodException])),
+            termNames_CONSTRUCTOR)
+          val exception = Apply(exceptionConstructor, List(msg))
+          Throw(exception)
+        }
+
+        def mkInvokeJUnitMethodRhs(
+            remainingTestMethods: List[(DefDef, Int)]): Tree = {
+          remainingTestMethods match {
+            case (method, id) :: tail =>
+              val symbol = method.symbol
+              val headMethodId = Literal(Constant(id.toString))
```

Comment:
I believe it is because at this phase the type `Int` has already been transformed to `Integer` and `int`. Therefore a bit more complicated to generate the method its bridge.